xtp/include/votca/xtp/checkpointreader.h

```diff
@@ -153,15 +153,11 @@ class CheckpointReader {
   const std::string path_;
   void ReadScalar(const CptLoc& loc, std::string& var,
                   const std::string& name) const {
-    const H5::DataType* strType = InferDataType<std::string>::get();
 
     H5::Attribute attr = loc.openAttribute(name);
+    H5::StrType stype = attr.getStrType();
 
-    H5std_string readbuf("");
-
-    attr.read(*strType, readbuf);
-
-    var = readbuf;
+    attr.read(stype, var);
```

Comment:
@JensWehner said: are we sure we do not actually have to resize it before passing it to read?

Yes, this is straight from the HDF5 H5cpp api examples. Also the `H5std_string` is just a different name for `std::string`, but has an extra option to use it without the `std`.